src/wasm/wasm-validator.cpp

```diff
@@ -701,31 +699,36 @@ void FunctionValidator::visitConst(Const* curr) {
 }
 
 void FunctionValidator::visitLocalGet(LocalGet* curr) {
-  shouldBeTrue(curr->index < getFunction()->getNumLocals(),
-               curr,
-               "local.get index must be small enough");
   shouldBeTrue(curr->type.isConcrete(),
                curr,
                "local.get must have a valid type - check what you provided "
                "when you constructed the node");
-  shouldBeTrue(curr->type == getFunction()->getLocalType(curr->index),
-               curr,
-               "local.get must have proper type");
+  if (shouldBeTrue(curr->index < getFunction()->getNumLocals(),
+                   curr,
+                   "local.get index must be small enough")) {
+    shouldBeTrue(curr->type == getFunction()->getLocalType(curr->index),
+                 curr,
+                 "local.get must have proper type");
+  }
 }
 
 void FunctionValidator::visitLocalSet(LocalSet* curr) {
-  shouldBeTrue(curr->index < getFunction()->getNumLocals(),
-               curr,
-               "local.set index must be small enough");
-  if (curr->value->type != unreachable) {
-    if (curr->type != none) { // tee is ok anyhow
-      shouldBeEqualOrFirstIsUnreachable(
-        curr->value->type, curr->type, curr, "local.set type must be correct");
+  if (shouldBeTrue(curr->index < getFunction()->getNumLocals(),
+                   curr,
+                   "local.set index must be small enough")) {
+    ;
+    if (curr->value->type != unreachable) {
+      if (curr->type != none) { // tee is ok anyhow
+        shouldBeEqualOrFirstIsUnreachable(curr->value->type,
+                                          curr->type,
+                                          curr,
+                                          "local.set type must be correct");
+      }
+      shouldBeEqual(getFunction()->getLocalType(curr->index),
+                    curr->value->type,
+                    curr,
+                    "local.set type must match function");
```

Comment:
Why move these into `if (shouldBeTrue(curr->index < getFunction()->getNumLocals() ...)`? It looks these can be checked independently from the local index failure.

Reply:
Because it is now an assertion failure to call `Function::getLocalType` with an out-of-bounds index.